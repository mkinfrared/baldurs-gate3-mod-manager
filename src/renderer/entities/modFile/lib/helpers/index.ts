export { extractPakFile } from "./extractPakFile";

export { getModFiles } from "./getModFiles";

export { getModInfo } from "./getModInfo";

export { readZipInfo } from "./readZipInfo";
