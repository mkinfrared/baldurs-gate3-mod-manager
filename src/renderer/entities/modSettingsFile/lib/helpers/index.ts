export { activateMod } from "./activateMod";

export { deactivateMod } from "./deactivateMod";

export { getCurrentSettings } from "./getCurrentSettings";

export { getInstalledMods } from "./getInstalledMods";

export { getModInfo } from "./getModInfo";

export { getNodeValue } from "./getNodeValue";
