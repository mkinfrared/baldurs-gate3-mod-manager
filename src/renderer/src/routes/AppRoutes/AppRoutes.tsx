import { Navigate, Route, Routes } from "react-router-dom";

import { BaldursGate3 } from "@renderer/pages";

import { APP_ROUTES } from "./AppRoutes.type";

const AppRoutes = () => (
  <Routes>
    <Route path={APP_ROUTES.BaldursGate3.path} element={<BaldursGate3 />} />
    <Route path="*" element={<Navigate to={APP_ROUTES.BaldursGate3.path} />} />
  </Routes>
);

export { AppRoutes };
