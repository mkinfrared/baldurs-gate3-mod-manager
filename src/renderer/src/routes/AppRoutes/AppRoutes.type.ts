import { route } from "react-router-typesafe-routes/dom";

export const APP_ROUTES = {
  BaldursGate3: route("baldurs-gate-3"),
};
