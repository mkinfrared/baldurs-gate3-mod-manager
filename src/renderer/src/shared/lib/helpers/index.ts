export { classNames } from "./classNames";

export { compose } from "./compose";

export { delay } from "./delay";

export { getNodeText } from "./getNodeText";

export { lazyImport } from "./lazyImport";

export { mergeRefs } from "./mergeRefs";

export { trpcClient, trpc } from "./trpc";

export * from "./typeGuards";
