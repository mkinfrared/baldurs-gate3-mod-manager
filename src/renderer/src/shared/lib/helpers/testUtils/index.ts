export { renderWithRouter, withQueryProvider } from "./testUtils";
