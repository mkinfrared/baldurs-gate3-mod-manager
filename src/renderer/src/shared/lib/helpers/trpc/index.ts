export { trpc, trpcClient } from "./trpc";
