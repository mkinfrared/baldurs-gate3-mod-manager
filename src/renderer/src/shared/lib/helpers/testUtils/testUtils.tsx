/* eslint-disable import/no-unused-modules */
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import { ReactElement, ReactNode } from "react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";

import { trpc, trpcClient } from "@renderer/shared/lib/helpers/trpc";

// https://testing-library.com/docs/example-react-router/
const renderWithRouter = (ui: ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};

const queryClient = new QueryClient();

const withQueryProvider = (children: ReactNode) => (
  <trpc.Provider client={trpcClient} queryClient={queryClient}>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </trpc.Provider>
);

const withRouter = (children: ReactNode) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export { renderWithRouter, withQueryProvider, withRouter };
