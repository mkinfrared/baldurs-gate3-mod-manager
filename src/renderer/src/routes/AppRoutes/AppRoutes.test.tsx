import { describe, expect, it, vi } from "vitest";

import { renderWithRouter } from "@renderer/shared/lib/helpers/testUtils";

import { AppRoutes } from "./AppRoutes";

vi.mock("@renderer/pages");

describe("<AppRoutes />", () => {
  const Component = <AppRoutes />;

  it("should be defined", () => {
    expect(AppRoutes).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = renderWithRouter(Component);

    expect(container).toMatchSnapshot();
  });
});
