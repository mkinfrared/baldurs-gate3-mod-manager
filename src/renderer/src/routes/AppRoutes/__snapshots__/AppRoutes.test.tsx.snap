// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AppRoutes /> > should match the snapshot 1`] = `<div />`;
